core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1020,6 +1036,17 @@ class ReplicaManager(val config: KafkaConfig,
             exception = None)
         }
       } catch {
+        case e@ (_: OffsetOutOfRangeExceptionWithOffsetValues) =>
+          LogReadResult(info = FetchDataInfo(LogOffsetMetadata.UnknownOffsetMetadata, MemoryRecords.EMPTY),
+              highWatermark = e.getHighWaterMark,
+              leaderLogStartOffset = e.getLogStartOffset,
+              leaderLogEndOffset = Log.UnknownOffset,
```

Comment:
nit: Why `UnknownOffset`?  Just add a one liner comment if this not needed as `highWatermark` and `LSO` are being set.

Reply:
Comment added